xdsl/dialects/qssa.py

```diff
@@ -0,0 +1,144 @@
+from abc import ABC
+
+from xdsl.dialects.builtin import IntegerAttr, IntegerType
+from xdsl.ir import Dialect, ParametrizedAttribute, SSAValue, TypeAttribute
+from xdsl.irdl import (
+    IRDLOperation,
+    VarOpResult,
+    irdl_attr_definition,
+    irdl_op_definition,
+    operand_def,
+    prop_def,
+    result_def,
+    var_result_def,
+)
+from xdsl.parser import Parser
+from xdsl.printer import Printer
+
+
+@irdl_attr_definition
+class QubitAttr(ParametrizedAttribute, TypeAttribute):
+    """
+    Type for a single qubit
+    """
+
+    name = "qssa.qubit"
+
+
+qubit = QubitAttr()
+
+
+class QubitBase(IRDLOperation, ABC):
+    pass
+
+
+@irdl_op_definition
+class QubitAllocOp(QubitBase):
+    name = "qssa.alloc"
+
+    num_qubits = prop_def(IntegerAttr)
+
+    res: VarOpResult = var_result_def(qubit)
+
+    def __init__(self, num_qubits: int):
+        super().__init__(
+            operands=(),
+            result_types=([qubit for _ in range(0, num_qubits)],),
+            properties={"num_qubits": IntegerAttr(num_qubits, 64)},
+        )
+
+    @classmethod
+    def parse(cls, parser: Parser) -> "QubitAllocOp":
+        with parser.in_angle_brackets():
+            num_qubits = parser.parse_integer()
+        attr_dict = parser.parse_optional_attr_dict()
+        return QubitAllocOp.create(
+            operands=(),
+            result_types=tuple(qubit for _ in range(0, num_qubits)),
+            properties={"num_qubits": IntegerAttr(num_qubits, 64)},
+            attributes=attr_dict or {},
+        )
+
+    def print(self, printer: Printer):
+        with printer.in_angle_brackets():
+            printer.print(self.num_qubits.value.data)
```

Comment:
```suggestion
            printer.print(len(self.res))
```
One approach would be to still print the number (since you're not printing a type) but not to store it on the op as a property.